developer-docs-site/docs/tutorials/full-node/network-identity-fullnode.md

```diff
@@ -0,0 +1,180 @@
+---
+title: "Network Identity For FullNode"
+slug: "network-identity-fullnode"
+sidebar_position: 12
+---
+
+# Network Identity For FullNode
+
+This guide will show you how to:
+- Create a static network identity for your FullNode.
+- Retrieve the public network identity.
+- Start a node with (or without) a static network identity.
+
+## Before you proceed
+
+Before you proceed, make sure that you already know how to start your local FullNode. See [Run a FullNode](run-a-fullnode) for detailed documentation.
+
+:::caution Docker support only on Linux
+
+Docker container is currently supported only on Linux x86-64 platform. If you are on macOS or Windows platform, use the Aptos-core source approach.
+
+:::
+
+## Creating a static identity for a FullNode
+
+FullNodes will automatically start up with a randomly generated network identity (a `PeerId` and a public key pair). This works well for regular FullNodes, but you may wish to be added to another node's allowlist, provide specific permissions or run your FullNode with the same identity. In this case, creating a static network identity can help.
+
+1. First, build the `aptos-operational-tool` using the [aptos-labs/aptos-core](https://github.com/aptos-labs/aptos-core) repo.
+
+    - Fork and clone the [aptos-labs/aptos-core](https://github.com/aptos-labs/aptos-core) repo. For example:
+
+    ```
+    $ git clone https://github.com/<YOUR-GITHUB-USERID>/aptos-core.git
+    $ cd aptos-core
+    $ ./scripts/dev_setup.sh
+    $ source ~/.cargo/env
+    ```
+
+    **Using Docker**
+
+    Alternatively, if you are on Linux x86-64 platform, you can use Aptos Docker image. `cd` into the directory for your local public FullNode and start a Docker container with the latest tools, for example:
+
+    ```
+    $ cd ~/my-full-node
+    $ docker run -it aptoslab/tools:devnet /bin/bash
+    ```
+
+2. Run the key generator to produce a hex encoded static x25519 private key. This will be the private key for your network identity.
+
+   ```
+    cargo run -p aptos-operational-tool -- \
+        generate-key \
+        --encoding hex \
+        --key-type x25519 \
+        --key-file /path/to/private-key.txt
+   ```
```

Comment:
We can actually replace these with the new tooling!  I forgot all about this.

You can generate a tool using the Aptos CLI:
```
aptos key generate --key-type x25519 --output-file /path/to/private-key.txt
```